src/Http/Event/ImportEventRequestHandler.php

```diff
@@ -152,13 +152,18 @@ public function handle(ServerRequestInterface $request): ResponseInterface
             }
 
             $this->aggregateRepository->save($eventAggregate);
+
+            $commands[] = new UpdateAttendanceMode($eventId, $event->getAttendanceMode());
         } else {
             if ($workflowStatus->sameAs(WorkflowStatus::READY_FOR_VALIDATION())) {
                 $commands[] = new Publish($eventId, $publishDate);
             }
 
             $commands[] = new UpdateTitle($eventId, $mainLanguage, $title);
             $commands[] = new UpdateType($eventId, $type->getId());
+            // The attendance mode needs to be updated before the location can be changed.
+            // For example passing a real location to an online event is not allowed.
+            $commands[] = new UpdateAttendanceMode($eventId, $event->getAttendanceMode());
```

Comment:
Do we check that the new location exists in the validation layer? If not, I think it's possible that we update the attendanceMode from online to offline/mixed, but then the `UpdateLocation()` after it might fail and the event might become offline/mixed but with a virtual location?

Is it possible to combine the update of the attendance mode and the update of the location in a single command, so if either one fails the other does not get changed?

Reply:
Validating the existence of the location is indeed missing here. We could take the same approach as in `UpdateLocationRequestHandler` and verify the existence of the location before doing any commands?

I think the check for and existing location is useful also out of the context of updating the attendance mode.

Reply:
Sounds good!

Reply:
I added validation to check the existence of the provided location.